junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -394,7 +407,8 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
 
 				@Override
 				public FileVisitResult visitFileFailed(Path file, IOException exc) {
-					if (exc instanceof NoSuchFileException) {
+					LOGGER.trace(exc, () -> "visitFileFailed: " + file);
+					if (exc instanceof NoSuchFileException && !Files.exists(file, LinkOption.NOFOLLOW_LINKS)) {
```

Comment:
:speech_balloon: This line fixes #4299

Reply:
It fixes the error, but it won't have deleted the junction, right?
Wouldn't it be possible to improve the implementation of `fileOperations.delete()` to not throw this error?

Reply:
> It fixes the error, but it won't have deleted the junction, right?

It will be deleted by the call to `resetPermissionsAndTryToDeleteAgain` below.

> Wouldn't it be possible to improve the implementation of `fileOperations.delete()` to not throw this error?

`FileOperations` exists only so that it can be replaced in tests.